spring-data-neo4j-rx/src/main/java/org/neo4j/springframework/data/core/schema/IdDescription.java

```diff
@@ -33,42 +34,70 @@
 public final class IdDescription {
 
 	/**
-	 * The selected strategy.
+	 * The class representing a generator for new ids or null for assigned ids.
 	 */
-	private final Id.Strategy idStrategy;
-
-	/**
-	 * The class representing a generator for new ids.
-	 */
-	private @Nullable final Class<? extends IdGenerator> idGeneratorClass;
+	private @Nullable final Class<? extends IdGenerator<?>> idGeneratorClass;
 
 	/**
 	 * The property that stores the id if applicable.
 	 */
 	private @Nullable final String graphPropertyName;
 
-	public IdDescription() {
-		this(Id.Strategy.INTERNALLY_GENERATED, null, null);
+	public static IdDescription forAssignedIds(String graphPropertyName) {
+
+		Assert.notNull(graphPropertyName, "Graph property name is required.");
+		return new IdDescription(null, graphPropertyName);
+	}
+
+	public static IdDescription forInternallyGeneratedIds() {
+		return new IdDescription(GeneratedValue.InternalIdGenerator.class, null);
 	}
 
-	public IdDescription(Id.Strategy idStrategy,
-		@Nullable Class<? extends IdGenerator> idGeneratorClass, @Nullable String graphPropertyName) {
-		this.idStrategy = idStrategy;
+	public static IdDescription forExternallyGeneratedIds(Class<? extends IdGenerator<?>> idGeneratorClass,
+		String graphPropertyName) {
+
+		Assert.notNull(idGeneratorClass, "Class of id generator is required.");
+		Assert.isTrue(idGeneratorClass != GeneratedValue.InternalIdGenerator.class,
+			"Cannot use InternalIdGenerator for externally generated ids.");
+		Assert.notNull(graphPropertyName, "Graph property name is required.");
+
+		return new IdDescription(idGeneratorClass, graphPropertyName);
+	}
+
+	private IdDescription(Class<? extends IdGenerator<?>> idGeneratorClass,
+		@Nullable String graphPropertyName) {
 		this.idGeneratorClass = idGeneratorClass;
 		this.graphPropertyName = graphPropertyName;
 	}
 
-	public Id.Strategy getIdStrategy() {
-		return idStrategy;
+	public Optional<Class<? extends IdGenerator<?>>> getIdGeneratorClass() {
+		return Optional.ofNullable(idGeneratorClass);
+	}
+
+	/**
+	 * @return True, if the database generated the ID.
+	 */
+	public boolean idIsGeneratedInternally() {
+		return this.idGeneratorClass == GeneratedValue.InternalIdGenerator.class;
 	}
 
-	public Optional<Class<? extends IdGenerator>> getIdGeneratorClass() {
-		return Optional.ofNullable(idGeneratorClass);
+	/**
+	 * @return True, if the ID is assigned to the entity before the entity hits the database, either manually or through a generator.
+	 */
+	public boolean idIsAssigned() {
```

Comment:
ould we name it something like `isIdAssigned` or `isAssignedId`? Sample applies to the sillibing methods.

Reply:
agree